pkgs/development/python-modules/plyer/default.nix

```diff
@@ -0,0 +1,21 @@
+{ lib, fetchPypi, buildPythonPackage, }:
+
+buildPythonPackage rec {
+  pname = "plyer";
+  version = "2.0.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-jeOLF7xDjfNu7erNVGzwUwOrhxhVrwZp/E6N9R8q35Q=";
+  };
+
+  doCheck = false; # tests fail in Nix sandbox
```

Comment:
How does it fail?

Reply:
If I add `keyring` as a check input, it fails with the following. I don't understand the internals, but passing in `xbacklight` does not help.

```
@nix { "action": "setPhase", "phase": "setuptoolsCheckPhase" }
setuptoolsCheckPhase
Executing setuptoolsCheckPhase
running test
WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox.
running egg_info
writing plyer.egg-info/PKG-INFO
writing dependency_links to plyer.egg-info/dependency_links.txt
writing requirements to plyer.egg-info/requires.txt
writing top-level names to plyer.egg-info/top_level.txt
reading manifest file 'plyer.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching '*README.rst'
writing manifest file 'plyer.egg-info/SOURCES.txt'
running build_ext
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.audio'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.barometer'
upower not found.Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.bluetooth'
sh: which: not found
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 94, in _ensure_obj
    obj = mod.instance()
  File "/build/plyer-2.0.0/plyer/platforms/linux/brightness.py", line 29, in instance
    return LinuxBrightness()
  File "/build/plyer-2.0.0/plyer/platforms/linux/brightness.py", line 18, in __init__
    raise Exception(msg)
Exception: It looks like 'xbacklight' is not installed. Try installing it with your distribution's package manager.
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.call'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.camera'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.compass'
xdg-open not found.Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.flash'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.gps'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.gravity'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.gyroscope'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.humidity'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.irblaster'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.light'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.proximity'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.sms'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.spatialorientation'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.stt'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.temperature'
lshw not found.Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
ModuleNotFoundError: No module named 'plyer.platforms.linux.vibrator'
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/platforms/linux/wifi.py", line 13, in <module>
    import wifi
ModuleNotFoundError: No module named 'wifi'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/utils.py", line 93, in _ensure_obj
    mod = __import__(module, fromlist='.')
  File "/build/plyer-2.0.0/plyer/platforms/linux/wifi.py", line 15, in <module>
    raise ModuleNotFoundError(
ModuleNotFoundError: python-wifi not installed. try: `pip install --user wifi`.
Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/platforms/win/uniqueid.py", line 6, in <module>
    import _winreg as regedit
ModuleNotFoundError: No module named '_winreg'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/build/plyer-2.0.0/plyer/platforms/win/uniqueid.py", line 9, in <module>
    import winreg as regedit
ModuleNotFoundError: No module named 'winreg'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 56, in <module>
    setup(
  File "/nix/store/z9rg0ff8qdmvg7v83fq0ixzp2vlpl3fl-python3.8-setuptools-50.3.1/lib/python3.8/site-packages/setuptools/__init__.py", line 153, in setup
    return distutils.core.setup(**attrs)
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/nix/store/z9rg0ff8qdmvg7v83fq0ixzp2vlpl3fl-python3.8-setuptools-50.3.1/lib/python3.8/site-packages/setuptools/command/test.py", line 232, in run
    self.run_tests()
  File "/nix/store/z9rg0ff8qdmvg7v83fq0ixzp2vlpl3fl-python3.8-setuptools-50.3.1/lib/python3.8/site-packages/setuptools/command/test.py", line 250, in run_tests
    test = unittest.main(
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/unittest/main.py", line 100, in __init__
    self.parseArgs(argv)
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/unittest/main.py", line 124, in parseArgs
    self._do_discovery(argv[2:])
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/unittest/main.py", line 244, in _do_discovery
    self.createTests(from_discovery=True, Loader=Loader)
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/unittest/main.py", line 154, in createTests
    self.test = loader.discover(self.start, self.pattern, self.top)
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/unittest/loader.py", line 349, in discover
    tests = list(self._find_tests(start_dir, pattern))
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/unittest/loader.py", line 405, in _find_tests
    tests, should_recurse = self._find_test_path(
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/unittest/loader.py", line 483, in _find_test_path
    tests = self.loadTestsFromModule(package, pattern=pattern)
  File "/nix/store/z9rg0ff8qdmvg7v83fq0ixzp2vlpl3fl-python3.8-setuptools-50.3.1/lib/python3.8/site-packages/setuptools/command/test.py", line 50, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/unittest/loader.py", line 191, in loadTestsFromName
    return self.loadTestsFromModule(obj)
  File "/nix/store/z9rg0ff8qdmvg7v83fq0ixzp2vlpl3fl-python3.8-setuptools-50.3.1/lib/python3.8/site-packages/setuptools/command/test.py", line 50, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/unittest/loader.py", line 191, in loadTestsFromName
    return self.loadTestsFromModule(obj)
  File "/nix/store/z9rg0ff8qdmvg7v83fq0ixzp2vlpl3fl-python3.8-setuptools-50.3.1/lib/python3.8/site-packages/setuptools/command/test.py", line 50, in loadTestsFromModule
    tests.append(self.loadTestsFromName(submodule))
  File "/nix/store/yl69v76azrz4daiqksrhb8nnmdiqdjg9-python3-3.8.8/lib/python3.8/unittest/loader.py", line 154, in loadTestsFromName
    module = __import__(module_name)
  File "/build/plyer-2.0.0/plyer/platforms/win/uniqueid.py", line 11, in <module>
    raise NotImplementedError()
NotImplementedError

```

Reply:
That does not look good. Did you check if the module still works?